exercises/non-retryable-error-types/README.md

```diff
@@ -0,0 +1,194 @@
+   the exception, verifying that the exception is no longer non-retryable.
+   Terminate this Workflow in the WebUI, as it will never successfully complete.
+6. Stop your Worker by using **Ctrl-C** in the terminal it is running in.
+
+## Part B: Configure Retry Policies to set Non-Retryable Error Types
+
+Now that the exception from the `ProcessCreditCard` Activity is no longer set to
+non-retryable, anyone who is running your Activity code may decide how to handle
+the failure. In this case, imagine you have decided that you do not want the
+Activity to retry upon failure -- but you don't want to "hardcode" it to
+non-retryable as in Exercise 1. In this part of the exercise, you will configure
+a Retry Policy to disallow this using non-retryable error types.
+
+Recall that a Retry Policy has the following attributes:
+
+- Initial Interval: Amount of time that must elapse before the first retry occurs
+- Backoff Coefficient: How much the retry interval increases (default is 2.0)
+- Maximum Interval: The maximum interval between retries
+- Maximum Attempts: The maximum number of execution attempts that can be made in the presence of failures
+
+You can also specify errors types that are not retryable in the Retry Policy.
+These are known as non-retryable error types. In the `ProcessCreditCard`
+Activity in `activities.go`, you returned an `ApplicationFailure` with the
+type `CreditCardError`. Now you will specify that error type as
+non-retryable.
+
+1. Open `workflow.go`. A `RetryPolicy` has already been defined with the
+   following configuration:
+
+```go
+	retrypolicy := &temporal.RetryPolicy{
+		MaximumInterval: time.Second * 10,
+		MaximumAttempts: 3,
+	}
+
+	options := workflow.ActivityOptions{
+		StartToCloseTimeout: time.Second * 120,
+		RetryPolicy:         retrypolicy,
+	}
+```
+
+2. Add an additional parameter to `retrypolicy` like so:
+
+```go
+NonRetryableErrorTypes: []string{"CreditCardError"},
+```
+
+3. Save and close the file.
+4. Verify that your Error is once again failing the Workflow. In one terminal,
+   start the Worker by running:
+   ```bash
+   go run worker/main.go
+   ```
+5. In another terminal, start the Workflow by executing `start/main.go` with bad
+   data to cause the error:
+   ```bash
+   go run start/main.go --creditcard 1234
+   ```
+6. Go to the WebUI and view the status of the Workflow. You should see an
+   `ActivityTaskFailed` error.
+
+## Part C: Add Heartbeats
+
+In this part of the exercise, you will add heartbeating to a new Activity,
+`PollDeliveryDriver` The `PollDeliveryDriver` method attempts to contact a
+driver to deliver the customers pizza. It may take a while for a delivery driver
+to accept the delivery, and you want to ensure that the Activity is still alive
+and processing. Heartbeats are used to do this, and fail fast if a failure is
+detected.
+
+In this exercise, instead of attempting to call an external service, success of
+the `PollDeliveryDriver` method call will be simulated.
```

Comment:
Personally, 'success of
the `PollDeliveryDriver` method call will be simulated' sounded off to me. Maybe something like, 'we will simulate a successful call of
the `PollDeliveryDriver` method.'